src/Functions/array/arrayAggregation.cpp

```diff
@@ -185,6 +192,22 @@ struct ArrayAggregateImpl
                         res[i] = x;
                     }
                 }
+                else if constexpr (aggregate_operation == AggregateOperation::product)
+                {
+                    size_t array_size = offsets[i] - pos;
+                    AggregationType product = 1;
+                    for (i = 0; i < array_size; ++i)
+                        product = product * x;
+
+                    if constexpr (IsDecimalNumber<Element>)
+                    {
+                        res[i] = DecimalUtils::convertTo<Result>(product, data.getScale() * array_size);
```

Comment:
Why there is `data.getScale() * array_size` ? Should not it be `data.getScale()` ?

Reply:
No, I have test this, the scale should be `* array_size`, otherwise the result is wrong.
eg: scale = 3, arrayProduct([1.000, 2.000, 3.000]) , if using `data.getScale()` the result = 6000000